src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -25940,6 +25942,190 @@ public void Add(MyCollection c)
+                ("System.Collections.Generic.ICollection<T>.get_Count()", "System.Collections.Generic.ICollection<T>.Count.get"),
+                ("System.Collections.Generic.ICollection<T>.get_IsReadOnly()", "System.Collections.Generic.ICollection<T>.IsReadOnly.get"),
+
+                ("System.Collections.Generic.IEnumerable<T>.GetEnumerator()", null),
+
+                ("System.Collections.Generic.IList<T>.IndexOf(T)", null),
+                ("System.Collections.Generic.IList<T>.Insert(int, T)", null),
+                ("System.Collections.Generic.IList<T>.RemoveAt(int)", null),
+                ("System.Collections.Generic.IList<T>.get_Item(int)", "System.Collections.Generic.IList<T>.this[int].get"),
+                ("System.Collections.Generic.IList<T>.set_Item(int, T)", "System.Collections.Generic.IList<T>.this[int].set"),
+
+                ("System.Collections.Generic.IReadOnlyCollection<T>.get_Count()", "System.Collections.Generic.IReadOnlyCollection<T>.Count.get"),
+                ("System.Collections.Generic.IReadOnlyList<T>.get_Item(int)", "System.Collections.Generic.IReadOnlyList<T>.this[int].get"),
+
+                ("System.Collections.ICollection.CopyTo(System.Array, int)", null),
+                ("System.Collections.ICollection.get_Count()", "System.Collections.ICollection.Count.get"),
+                ("System.Collections.ICollection.get_IsSynchronized()", "System.Collections.ICollection.IsSynchronized.get"),
+                ("System.Collections.ICollection.get_SyncRoot()", "System.Collections.ICollection.SyncRoot.get"),
+
+                ("System.Collections.IEnumerable.GetEnumerator()", null),
+
+                ("System.Collections.IList.Add(object)", null),
+                ("System.Collections.IList.Clear()", null),
+                ("System.Collections.IList.Contains(object)", null),
+                ("System.Collections.IList.IndexOf(object)", null),
+                ("System.Collections.IList.Insert(int, object)", null),
+                ("System.Collections.IList.Remove(object)", null),
+                ("System.Collections.IList.RemoveAt(int)", null),
+                ("System.Collections.IList.get_IsFixedSize()", "System.Collections.IList.IsFixedSize.get"),
+                ("System.Collections.IList.get_IsReadOnly()", "System.Collections.IList.IsReadOnly.get"),
+                ("System.Collections.IList.get_Item(int)", "System.Collections.IList.this[int].get"),
+                ("System.Collections.IList.set_Item(int, object)", "System.Collections.IList.this[int].set"),
+            };
+
+            foreach (var (memberName, sourceName) in members)
+            {
+                string expectedIl = verifier.VisualizeIL($"ReadOnlySingletonList<T>.{sourceName ?? memberName}")
+                    .Replace("ReadOnlySingletonList<T>", "<>z__ReadOnlySingletonList<T>");
+                verifier.VerifyIL($"<>z__ReadOnlySingletonList<T>.{memberName}", expectedIl);
+            }
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

We should include a test that executes each of these interface members. Consider adding a test similar to the existing `SynthesizedReadOnlyArray` test, at least up `CompileAndVerify(..., expectedOutput="...")`. (Since we're verifying the IL for each of the members in this test, it's probably not necessary to verify the IL in the new test.)